backend/src/role.rs

```diff
@@ -19,6 +31,44 @@ pub enum RoleError {
     Unauthorized,
 }
 
+#[get("/<role_id>/questions")]
+pub async fn get_question_ids(
+    role_id: i32,
+    user: User,
+    db: Database,
+) -> Result<Json<QuestionIdsResponse>, Json<QuestionsError>> {
+    // First check that the role is valid and the user should be able to access the ids.
+    // We can't use the helper function below since behaviour depends on the draft
+    // status of the campaign.
+
+    let role = db
+        .run(move |conn| Role::get_from_id(conn, role_id))
+        .await
+        .ok_or(QuestionsError::RoleNotFound)?;
+
+    let campaign = db
+        .run(move |conn| Campaign::get_from_id(conn, role.campaign_id))
+        .await
+        .ok_or(QuestionsError::CampaignNotFound)?;
+
+    let org_user = db
+        .run(move |conn| OrganisationUser::get(conn, campaign.organisation_id, user.id))
+        .await
+        .ok_or(QuestionsError::Unauthorized)?;
+
+    // Prevent people from viewing while it's in draft mode,
+    // unless they have adequate permissions
+    if campaign.draft && !user.superuser && org_user.admin_level == AdminLevel::ReadOnly {
```

Comment:
You can probably replace it with

https://github.com/csesoc/chaos/blob/main/backend/src/database/models.rs#L79

which is our helper type for construcitng orfganisation directors/admins

Reply:
In this case we would still need to get the campaign separately to check whether it's in draft mode (if not draft, anyone should be able to access) - means multiple accesses of the campaign in the db. It would probably be cached, but I don't know much about db performance. I will change it to use the helper type + separately check draft status if this is ok though.

Reply:
hmm that is a fair point

I think your suggestion of helper + seperate draft is a good idea